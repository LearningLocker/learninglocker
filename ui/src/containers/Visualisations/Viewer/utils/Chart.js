// @ts-check
import React from 'react';
import { LineChart, XAxis, YAxis, Bar, Legend, Tooltip } from 'recharts';
import { union } from 'lodash';
import { AutoSizer } from 'react-virtualized';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
// @ts-ignore
import styles from './styles.css';

/**
 * @typedef {Object} AutoSizerProps
 * @property {number} height
 * @property {number} width
 */

const render = withStyles(styles)(
  /**  @param {{ xAxisLabel: string, yAxisLabel: string, children: (props: AutoSizerProps) => React.Element }} props */
  (props) => {
    const { xAxisLabel, yAxisLabel, children } = props;
    return (
      <div className={styles.chart}>
        <div className={styles.xAxisLabel}>
          <span className={styles.xAxis}>
            {xAxisLabel || 'X Axis'}
          </span>
        </div>
        <div className={`${styles.barContainer}`}>
          <div className={styles.yAxisLabel}>
            <span className={styles.yAxis}>
              {yAxisLabel || 'Y Axis'}
            </span>
          </div>
          <div className={styles.chartWrapper}>
            <AutoSizer>
              {children}
            </AutoSizer>
          </div>
        </div>
      </div>
    );
  }
);

export default render;
