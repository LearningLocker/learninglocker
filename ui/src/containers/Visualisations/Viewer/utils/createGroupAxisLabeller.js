// @ts-check

/**
 * @param {{ [groupId: string ]: string }} groupDictionary
 * @returns {(groupId: string) => string}
 */
const createGroupAxisLabeller = (groupDictionary) => {
  return (groupId) => {
    const name = groupDictionary[groupId];
    return name !== undefined ? name : groupId;
  };
};

export default createGroupAxisLabeller;
