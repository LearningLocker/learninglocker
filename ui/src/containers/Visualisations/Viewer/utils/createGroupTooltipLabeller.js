// @ts-check

/**
 * @param {{ [groupId: string ]: string }} groupDictionary
 * @returns {(groupId: string) => string}
 */
const createGroupTooltipLabeller = (groupDictionary) => {
  return (groupId) => {
    const name = groupDictionary[groupId];
    return name !== undefined ? name : groupId;
  };
};

export default createGroupTooltipLabeller;
