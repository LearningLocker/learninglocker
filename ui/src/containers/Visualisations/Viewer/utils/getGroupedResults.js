// @ts-check

/**
 * @typedef {Object} GroupResult
 * @property {string} _id
 * @property {string} model
 * @property {number} count
 */

/**
 * @param {GroupResult[]} groupResults
 * @returns {{[groupId: string]: { groupId: string, name: string, count: number }}}
 */
const getGroupedResults = (groupResults) => {
  return groupResults.reduce((groupedSeriesResult, groupResult) => {
    const { _id: groupId, model: name, count } = groupResult;
    groupedSeriesResult[groupResult._id] = { groupId, name, count };
    return groupedSeriesResult;
  }, {});
};

export default getGroupedResults;
