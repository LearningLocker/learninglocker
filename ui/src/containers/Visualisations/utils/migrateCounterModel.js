export default (model) => {
  return {
    organisation: model.get('organisation'),
    owner: model.get('owner'),
    description: model.get('description'),
    isPublic: model.get('isPublic'),
    type: model.get('type'),
    config: {
      previewPeriod: model.get('previewPeriod'),
      filter: model.getIn(['filters', 0]),
      value: {
        key: model.getIn(['axesvalue', 'optionKey']),
        operator: model.get('axesoperator'),
        colour: 'red',
      },
    },
    createdAt: model.get('createdAt'),
    updatedAt: model.get('updatedAt'),
    _id: model.get('_id'),
  };
};
