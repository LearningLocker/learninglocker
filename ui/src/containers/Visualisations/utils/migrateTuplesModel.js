export default (model) => {
  return {
    organisation: model.get('organisation'),
    owner: model.get('owner'),
    description: model.get('description'),
    isPublic: model.get('isPublic'),
    type: model.get('type'),
    config: {
      previewPeriod: model.get('previewPeriod'),
      series: model.get('filters').map((series, seriesIndex) => {
        return {
          label: series.get('label', `Series ${seriesIndex}`),
          colour: series.get('color', 'orange'),
          filter: series.get('filter'),
        };
      }).toJS(),
      group: {
        label: model.getIn(['axesgroup', 'searchString']),
        key: model.getIn(['axesgroup', 'optionKey']),
      },
      firstValue: {
        label: model.getIn(['axesxValue', 'searchString']),
        key: model.getIn(['axesxValue', 'optionKey']),
        operator: model.get('axesxOperator'),
        filter: model.get('axesxQuery'),
      },
      secondValue: {
        label: model.getIn(['axesyValue', 'searchString']),
        key: model.getIn(['axesyValue', 'optionKey']),
        operator: model.get('axesyOperator'),
        filter: model.get('axesyQuery'),
      },
    },
    createdAt: model.get('createdAt'),
    updatedAt: model.get('updatedAt'),
    _id: model.get('_id'),
  };
};
