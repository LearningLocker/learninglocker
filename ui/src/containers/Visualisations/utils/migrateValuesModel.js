export default (model) => {
  return {
    organisation: model.get('organisation'),
    owner: model.get('owner'),
    description: model.get('description'),
    isPublic: model.get('isPublic'),
    type: model.get('type'),
    config: {
      previewPeriod: model.get('previewPeriod'),
      stacked: model.get('stacked'),
      series: model.get('filters').map((series, seriesIndex) => {
        return {
          label: series.get('label', `Series ${seriesIndex}`),
          colour: series.get('color', 'orange'),
          filter: series.get('filter'),
        };
      }).toJS(),
      group: {
        label: model.getIn(['axesgroup', 'searchString']),
        key: model.getIn(['axesgroup', 'optionKey']),
      },
      value: {
        label: model.getIn(['axesvalue', 'searchString']),
        key: model.getIn(['axesvalue', 'optionKey']),
        operator: model.get('axesoperator'),
      },
    },
    createdAt: model.get('createdAt'),
    updatedAt: model.get('updatedAt'),
    _id: model.get('_id'),
  };
};
