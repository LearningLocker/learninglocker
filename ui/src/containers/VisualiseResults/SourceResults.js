import React from 'react';
import { compose } from 'recompose';
import { OrderedMap } from 'immutable';
import { withStatementsVisualisation } from 'ui/utils/hocs';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import styles from './styles.css';

const getAxes = (index, axes) => {
  switch (index) {
    case 0:
      return axes.get('xLabel', 'X-Axes');
    case 1:
      return axes.get('yLabel', 'Y-Axes');
    default:
      return axes.get('xLabel', 'X-Axes');
  }
};

export const generateTableData = (results, labels, axes) => {
  const seriesList = labels.zip(results);
  const seriesList2 = seriesList.map(([key, item], i) => {
    if (key === undefined) {
      return [`s${i}`, item];
    }
    return [key, item];
  });

  const seriesMap = new OrderedMap(seriesList2);

  const result = seriesMap.reduce((reduction, series, seriesKey) =>

    series.reduce((axesReduction, axes2, axesKey) =>

      axes2.reduce((seriesReduction, item) => {

        return seriesReduction.setIn(
          [item.get('model'), seriesKey, getAxes(axesKey, axes)],
          item
        );
      }, axesReduction)
    , reduction)
  , new OrderedMap());

  return result;
};

export default compose(
  withStatementsVisualisation,
  withStyles(styles)
)(({
  getFormattedResults,
  results,
  labels,
  axes
}) => {
  const formattedResults = getFormattedResults(results);

  const tableData = generateTableData(formattedResults, labels, axes);

  return (
    <div className={styles.sourceResultsContainer}>
      <table className="table table-bordered table-striped">
        <tbody>
          <tr>
            <th />
            {tableData.first().map((item, key) => (
              <th colSpan={item.size}>{key}</th>
            ))}
          </tr>
          <tr>
            <th>i</th>
            {
              tableData.first().map(series => (
                series.map((axes2, axesKey) => (
                  <th>{axesKey}</th>
                ))
              ))
            }
          </tr>
          {tableData.map((item, key) => (
            <tr>
              <td>{key}</td>
              {item.map(series =>
                series.map(axes2 =>
                  (<td>{axes2.get('count')}</td>)
                )
              )}
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
});
