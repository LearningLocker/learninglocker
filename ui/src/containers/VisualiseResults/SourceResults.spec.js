import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { withInsertCSS } from 'ui/utils/hocs';
import SourceResults, { generateTableData } from './SourceResults';

const data = fromJS([[
  {
    0: {
      count: 5,
      model: 'monday'
    }
  }
]]);

const series = fromJS(['series 1']);

const axes = fromJS({ xLabel: 'the-x-label' });

test('SourceResults generate correct data', () => {
  const result = generateTableData(data, series, axes);

  expect(result.getIn(['monday', 'series 1', 'the-x-label', 'count'])).toEqual(5);
});

test('SourceResults should render', () => {
  const wrappedSourceResults = withInsertCSS(SourceResults);

  const criterion = ReactTestRenderer.create(
    <wrappedSourceResults
      results={data}
      labels={series}
      axes={axes} />
  ).toJSON();

  expect(criterion).toMatchSnapshot();
});
