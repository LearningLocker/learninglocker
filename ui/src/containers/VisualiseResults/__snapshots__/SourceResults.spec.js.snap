// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SourceResults should render 1`] = `
<wrappedSourceResults
  axes={
    Object {
      "xLabel": "the-x-label",
    }
  }
  labels={
    Array [
      "series 1",
    ]
  }
  results={
    Array [
      Array [
        Object {
          "0": Object {
            "count": 5,
            "model": "monday",
          },
        },
      ],
    ]
  }
/>
`;
